core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -52,6 +52,16 @@ private[spark] class CoarseGrainedExecutorBackend(
     driver ! RegisterExecutor(executorId, hostPort, cores)
     context.system.eventStream.subscribe(self, classOf[RemotingLifecycleEvent])
   }
+  
+  override def isValidMessage(msg: Any) = msg match {
+    case disEvt: DisassociatedEvent =>
+      if (disEvt.remoteAddress == driver.anchorPath.address) {
```

Comment:
no...it's hard to get a universal condition to check whether DisassociatedEvent is valid or not

e.g. for worker, the condition is "x.remoteAddress == masterAddress"

for master, x.remoteAddress is in driverAddresses || x.remoteAddress is in workerAddresses

In current implementation, Master/Worker/CoarseGrainedSchedulerBackend maintain in-memory data structures, which are luckily related to x.remoteAddress. So even they received DisassociatedEvent from irrelevant remoteAddresses, they just ignore that by running "RemoteAddress-Related-Map.get(remoteAddress).foreach{...}"

ExecutorBackend does not get that luck, so we have to check explicitly in the way proposed in this PR....

I made it as an abstract method in case we will introduce more actor-based components in future so that we can extend that